### ISSUE TITLE
CompassApp.run option to write responses to file system

### ISSUE THREAD

**Issue by robfitzgerald** [NREL/routee-compass#19]
when batches are submitted to CompassApp, it keeps all incremental response objects in memory until the batch is completed. for very large batches, this may result in an out-of-memory error. to avoid this, we can immediately write the response to disk and then discard the object from RAM.

implementation will require adding functionality to the rust CompassApp that writes responses to disk and removes the response object. this can be done one of two ways:
- add another run method to CompassApp that appends each response object from a chunk to some output file
- create a "to disk" output plugin _(rjf: i think i prefer this one, it doesn't require changes to the CompassApp Python or Rust API)_
  - perhaps a refactor of OutputPlugins to perform a `flat_map` operation similar to InputPlugins, so we can flatten the result

serializing responses as JSON and appending them to a file is likely simplest if we stick to a newline-delimited JSON format, and if we serialize the JSON as a single row (no newlines) object.

**Comment by nreinicke** [comment]
> create a "to disk" output plugin

Yeah I like this option as well, I'm wondering what the plugin should return the main application.. we could either:
 - return an empty json object
 - return a small json object with something like:

```json
{
  "request": ...,
  "status": "Successfully wrote result to file my_output_file.json" 
}
```

**Comment by robfitzgerald** [comment]
right, BUT! if we refactor OutputPlugin so that, instead of `map` ing over the responses, we instead `flat_map`, we can also return _nothing_. that will truly void out any memory accumulation. we can get to that by having OutputPlugins return a Vec<JSON> similar to [InputPlugin](https://github.com/NREL/routee-compass/blob/main/rust/routee-compass/src/plugin/input/input_plugin.rs):

```rust
pub trait OutputPlugin: Send + Sync {
    fn process(
        &self,
        output: &serde_json::Value,
        result: Result<&SearchAppResult, SearchError>,
    ) -> Result<Vec<serde_json::Value>, PluginError>;
}
```

**Comment by nreinicke** [comment]
Great point, yeah returning nothing seems to be ideal. I guess we would still return a vector of any requests that resulted in an error.
